### PR TITLE
sdl: add hadolint

### PR DESCRIPTION
**Pull request by NeethuES-intel**
### CODE MAINTAINABILITY

- [Y] **_Commit Message meets guidelines as indicated in the URL\*_**
    - <https://github.com/intel/AiCSD/blob/main/.github/CONTRIBUTING.md>
- [ ] Tests for the changes have been added (required for bug fixes / features). **If not, state why not**
- [ ] Added labels (these are git labels such as bug or enhancement)
- [ ] For any API changes or service changes, updates to the documentation were made (Swagger APIs & GitHub Docs). **If not, state why not**
- [ ] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**

### What are the steps to test this PR contribution?

How does a reviewer know it works?

Are there any special instructions beyond standard workflow?

### QUALITY CHECKS

- [ ] **_Quality of code (At least one should be checked as applicable)\*_**
    - [ ] Commit Message meets guidelines
    - [ ] Code copyright is correct
    - [ ] PR changes adhere to industry practices and standards as well as domain or language specific anti-patterns
    - [ ] Adopted domain specific coding standards ([Go Coding Standards](https://go.dev/doc/effective_go))
    - [ ] Code is adequately commented and documented
    - [ ] Confusing logic is explained in comments
    - [ ] Error and exception code paths implemented correctly
    - [ ] Tracing output are minimized and logic
- [ ] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [ ] **_Static code scan report shows zero critical issues\*_**

## Other information

### CODE REVIEW

**Review by ejlee3 — DISMISSED** [review]
LGTM

**Review by jim-wang-intel — COMMENTED** [review]
Hi @NeethuES-intel  
  This is just a config or policy-rule yaml file for hadolint to use while scanning the Dockerfile.
  Where is the installation of hadolint binary and the execution of hadolint on the github actions?  reference to automated-self-checkout: https://github.com/intel-retail/automated-self-checkout/blob/main/.github/workflows/reviewdog.yaml#L33-L38

**Review by antoniomtz — DISMISSED** [review]

**Comment by NeethuES-intel** [comment]
> Hi @NeethuES-intel This is just a config or policy-rule yaml file for hadolint to use while scanning the Dockerfile. Where is the installation of hadolint binary and the execution of hadolint on the github actions? reference to automated-self-checkout: https://github.com/intel-retail/automated-self-checkout/blob/main/.github/workflows/reviewdog.yaml#L33-L38

@jim-wang-intel thanks for pointing it out. I added the entire reviewdog.yaml file for scan.

File: .github/workflows/reviewdog.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      pull-requests: write
+    steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
+          with:
+            ref: ${{ github.event.pull_request.head.sha }}
+        - name: Install go
+          uses: actions/setup-go@v4
+          with:
+            go-version: '1.20'
```
Comment (ejlee3) [inline .github/workflows/reviewdog.yaml:19]: should this be 
```suggestion
            go-version: '1.21'
```

**Review by ejlee3 — APPROVED** [review]
LGTM

**Pull request merged**